gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -314,6 +355,10 @@ private ManagedChannel createSingleChannel() throws IOException {
             .intercept(metadataHandlerInterceptor)
             .userAgent(headerInterceptor.getUserAgentHeader())
             .executor(executor);
+    SslContext sslContext = createSslContext();
+    if (sslContext != null) {
+      builder = ((NettyChannelBuilder) builder).sslContext(sslContext);
```

Comment:
The only way this would be safe is if this builder was created with a method like `NettyChannelBuilder.forTarget()`. If you use `ManagedChannelBuilder.forTarget()` then there are no guarantees the type of object returned and that can change based on the environment or every process.

Reply:
I will re-implement this part when grpc-java provides a new stable api to add the client certificate to the channel builder.